x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/RollupShardIndexer.java

```diff
@@ -248,319 +191,203 @@ public void afterBulk(long executionId, BulkRequest request, Throwable failure)
             .build();
     }
 
-    private Rounding createRounding(RollupActionDateHistogramGroupConfig groupConfig) {
-        DateHistogramInterval interval = groupConfig.getInterval();
-        ZoneId zoneId = groupConfig.getTimeZone() != null ? ZoneId.of(groupConfig.getTimeZone()) : null;
-        Rounding.Builder tzRoundingBuilder;
-        if (groupConfig instanceof RollupActionDateHistogramGroupConfig.FixedInterval) {
-            TimeValue timeValue = TimeValue.parseTimeValue(interval.toString(), null, getClass().getSimpleName() + ".interval");
-            tzRoundingBuilder = Rounding.builder(timeValue);
-        } else if (groupConfig instanceof RollupActionDateHistogramGroupConfig.CalendarInterval) {
-            Rounding.DateTimeUnit dateTimeUnit = DateHistogramAggregationBuilder.DATE_FIELD_UNITS.get(interval.toString());
-            tzRoundingBuilder = Rounding.builder(dateTimeUnit);
-        } else {
-            throw new IllegalStateException("unsupported interval type");
-        }
-        return tzRoundingBuilder.timeZone(zoneId).build();
-    }
-
-    private void indexBucket(BucketKey key, List<FieldMetricsProducer> fieldsMetrics, int docCount) {
-        IndexRequestBuilder request = client.prepareIndex(tmpIndex);
-        Map<String, Object> doc = Maps.newMapWithExpectedSize(2 + key.groupFields.size() + fieldsMetrics.size());
-        doc.put(DocCountFieldMapper.NAME, docCount);
-        doc.put(timestampField.name(), timestampFormat.format(key.timestamp));
+    private class TimeSeriesBucketCollector extends BucketCollector {
+        private final BulkProcessor bulkProcessor;
+        private long docsProcessed;
+        private long bucketsCreated;
```

Comment:
I think it'd be useful to plumb these into the task so the get task API can show you the progress. It'd be fine to do in a follow up though.